### ISSUE TITLE
id-match onlyDeclarations evaluates property assignment value (when it should not)

### ISSUE THREAD

**Issue by gajus** [eslint/eslint#5885]
**What version of ESLint are you using?**

v2.8.0

**What parser (default, Babel-ESLint, etc.) are you using?**

babel-eslint

**Please show your full configuration:**

```json
{
    "rules": {
        "id-match": [
            2,
            "(^[A-Za-z]+(?:[A-Z][a-z]*)*\\d*$)|(^[A-Z]+(_[A-Z]+)*(_\\d$)*$)|(^(_|\\$)$)",
            {
                "onlyDeclarations": true,
                "properties": true
            }
        ]
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

The subject code:

```js
export default {
    context: __dirname
};
```

**What did you expect to happen?**

I expected no linting error.

**What actually happened? Please include the actual, raw output from ESLint.**

I got a linting error:

```
║ Line     │ Column   │ Type     │ Message                                                │ Rule ID              ║
╟──────────┼──────────┼──────────┼────────────────────────────────────────────────────────┼──────────────────────╢
║ 25       │ 14       │ error    │ Identifier '__dirname' does not match the pattern      │ id-match             ║
║          │          │          │ '(^[A-Za-z]+(?:[A-Z][a-z]*)*\d*$)|(^[A-Z]+(_[A-        │                      ║
║          │          │          │ Z]+)*(_\d$)*$)|(^(_|\$)$)'                             │                      ║
```

This is in relation to the following PR, https://github.com/eslint/eslint/pull/5631.

**Comment by gajus** [comment]
I don't understand what is the reason `__dirname` is treated as a property. This is the code bit that wrongly (?) assumes that `__dirname` is a property, https://github.com/gajus/eslint/blob/ccd8ca91c920f0a847c9510e6f83aa01aff2b07e/lib/rules/id-match.js#L97-L107.

**Comment by platinumazure** [comment]
I wonder if that bit of code was meant to check property keys only? In that case, it would be sufficient to check `node.parent.type === "Property" && node.parent.key === node`.

**Comment by kaicataldo** [comment]
It looks like it warns for both the keys and values of properties...Read over the rule docs, and it's not clear to me if that was the original intent or not. If we decide this is the desired behavior, we should make the docs clearer. @eslint/eslint-team thoughts?

**Comment by platinumazure** [comment]
I think it should be keys only. If someone wanted to flag property values, too, they could turn off onlyDeclarations so it flags uses of identifiers that are not declared in the same script or module.

**Comment by mikesherov** [comment]
Yeah, this just seems like a bug. I'd like to take this one.

**Comment by mikesherov** [comment]
Looks like https://github.com/eslint/eslint/pull/5631/files also missed adding the new option to the rule  schema. There's no automatic test for that? I suppose I should file a separate chore for that...

**Comment by mikesherov** [comment]
@kaicataldo care to review https://github.com/eslint/eslint/pull/5960 ?
